Docs - Changed documentation for VPC SC perimeter ingress policy sources

daanheikens opened the pull request:
Added documentation about the support of VPC sources for `IngressPolicy[].from.resource` fields

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
accesscontextmanager: Added docs for `google_access_context_manager_service_perimeter` and `google_access_context_manager_service_perimeter_ingress_policy` which reflects VPC network support for the `from.resource` field
```

daanheikens commented:
@SarahFrench just a note, i saw that all the VPC SC examples have the `skip_test` field to `true`. Why is that?  I wanted to check in why before making any tests, happy to do so by the way.

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccAccessContextManager`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/42ed69c2-89ce-408b-9a55-b5fc85d87230/recording/TestAccAccessContextManager.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/42ed69c2-89ce-408b-9a55-b5fc85d87230/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/42ed69c2-89ce-408b-9a55-b5fc85d87230/recording) for each test

SarahFrench commented on mmv1/products/accesscontextmanager/ServicePerimeter.yaml:
```diff
@@ -269,10 +269,10 @@ properties:
+                          Currently only projects and VPCs are allowed.
+                          Project format: projects/{projectNumber} VPC network format: //compute.googleapis.com/projects/{PROJECT_ID}/global/networks/{NAME}.
+                          The project may be in any Google Cloud organization, not just the organization that the perimeter is defined in.
+                          * is not allowed, the case of allowing all Google Cloud resources only is not supported.
```
Could you please make this change for both files, thanks!

SarahFrench reviewed (changes requested):
Hi, thanks for your PR! I've noticed an issue with the markdown generated from these changes, but once that's fixed I'm happy to merge

Edit: also, ignore the call to action to add more tests in [this message](https://github.com/GoogleCloudPlatform/magic-modules/pull/10806#issuecomment-2135541527)- those alerts are meant to ensure people adding new fields add test coverage, but in this case it's identifying issues from before this PR was opened.

daanheikens commented:
> Hi, thanks for your PR! I've noticed an issue with the markdown generated from these changes, but once that's fixed I'm happy to merge
> 
> 
> 
> Edit: also, ignore the call to action to add more tests in [this message](https://github.com/GoogleCloudPlatform/magic-modules/pull/10806#issuecomment-2135541527)- those alerts are meant to ensure people adding new fields add test coverage, but in this case it's identifying issues from before this PR was opened. 

Hi! Thanks for the helpful feedback, i will see to it!

SarahFrench commented:
>SarahFrench just a note, i saw that all the VPC SC examples have the skip_test field to true. Why is that? I wanted to check in why before making any tests, happy to do so by the way.

I'm not sure, but with help from git blame I found this comment thread : https://github.com/GoogleCloudPlatform/magic-modules/pull/1070/files#r242342225

Testing things at organization level can be hard in our test environment, but if you want to add acceptance tests and/or unskip those examples we might find that the restrictions present in 2018 have changed

daanheikens commented:
> > SarahFrench just a note, i saw that all the VPC SC examples have the skip_test field to true. Why is that? I wanted to check in why before making any tests, happy to do so by the way.
> 
> I'm not sure, but with help from git blame I found this comment thread : https://github.com/GoogleCloudPlatform/magic-modules/pull/1070/files#r242342225
> 
> Testing things at organization level can be hard in our test environment, but if you want to add acceptance tests and/or unskip those examples we might find that the restrictions present in 2018 have changed

@SarahFrench i removed the `skip_test` for both resource examples, could you run the tests to see if they work/if they show problems? I will create an example asap.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 6 files changed, 595 insertions(+), 16 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 6 files changed, 595 insertions(+), 16 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 20 files changed, 767 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (11 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}

```
Resource: `google_access_context_manager_service_perimeter_ingress_policy` (1 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter_ingress_policy" "primary" {
  ingress_from {
    sources {
      resource = # value needed
    }
  }
}

```

SarahFrench commented on mmv1/products/accesscontextmanager/ServicePerimeter.yaml:
```diff
@@ -54,26 +54,22 @@ mutex: '{{name}}'
     vars:
       access_level_name: 'secure_data_exchange'
   - !ruby/object:Provider::Terraform::Examples
     name: 'access_context_manager_service_perimeter_dry-run'
```
```suggestion
    name: 'access_context_manager_service_perimeter_dry_run'
```

Please rename the example file that this line is referencing to also have `_` instead of `-`

SarahFrench reviewed (changes requested):
The error after the tests were unskipped is:

```
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_generated_test.go:235:89: expected '(', found '-'
```

Looks like its because [the generated function names are malformed](https://github.com/modular-magician/terraform-provider-google-beta/blob/1ace1be3773d75f78844ed78001f93f9325f775e/google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_generated_test.go#L235C6-L235C100). I've left a review comment showing how to address this.

daanheikens commented:
> The error after the tests were unskipped is:
> 
> ```
> google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_generated_test.go:235:89: expected '(', found '-'
> ```
> 
> Looks like its because [the generated function names are malformed](https://github.com/modular-magician/terraform-provider-google-beta/blob/1ace1be3773d75f78844ed78001f93f9325f775e/google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_generated_test.go#L235C6-L235C100). I've left a review comment showing how to address this.

Adjusted! Also added examples of the `resource` argument in both resources

SarahFrench reviewed (changes requested):
Please push a commit to rename these files to have the new access_context_manager_service_perimeter_dry_run name:

- https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/access_context_manager_service_perimeter_dry-run.tf.erb
- https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/go/access_context_manager_service_perimeter_dry-run.tf.tmpl

daanheikens commented:
> Please push a commit to rename these files to have the new access_context_manager_service_perimeter_dry_run name:
> 
> * https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/access_context_manager_service_perimeter_dry-run.tf.erb
> * https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/go/access_context_manager_service_perimeter_dry-run.tf.tmpl

Not sure what went wrong, i thought i renamed at least the first one.... Pushed the name changes!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 6 files changed, 636 insertions(+), 17 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 6 files changed, 636 insertions(+), 17 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 20 files changed, 788 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics
Total tests: `0`
Passed tests: `0`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>accesscontextmanager</li></ul></blockquote></details>


#### Non-exercised tests

Tests were added that are skipped in VCR:

- TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample

$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/455d6463-874f-4a51-ad37-7c6c049a7c80/build-log/replaying_test.log)

daanheikens commented:
> This is the error reported from the automated tests on this PR:
> 
> ```
> Error: google/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go:48:6: testAccCheckAccessContextManagerServicePerimeterIngressPolicyDestroyProducer redeclared in this block
> Error: 	google/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_generated_test.go:1[16](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/9317884630/job/25649106399#step:8:17):6: other declaration of testAccCheckAccessContextManagerServicePerimeterIngressPolicyDestroyProducer
> Error: google/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:92:6: testAccCheckAccessContextManagerServicePerimeterDestroyProducer redeclared in this block
> Error: 	google/services/accesscontextmanager/resource_access_context_manager_service_perimeter_generated_test.go:383:6: other declaration of testAccCheckAccessContextManagerServicePerimeterDestroyProducer
> ```
> 
> The [handwritten tests have a `testAccCheckAccessContextManagerServicePerimeterDestroyProducer` function](https://github.com/GoogleCloudPlatform/magic-modules/blob/78c55099c2b854c121c17080a66bdcba68d91065/mmv1/third_party/terraform/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go#L46), and the generated test code is [adding a second declaration of that function](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806#diff-81080c5670e7dc23c8cf8963ae5267f2bf1c466647365b4281d57d48be38aa38R116).
> 
> You should [remove this function](https://github.com/GoogleCloudPlatform/magic-modules/blob/78c55099c2b854c121c17080a66bdcba68d91065/mmv1/third_party/terraform/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go#L46) to address the build error. Having said that, though, the handwritten and generated version of those functions are different. We'd want to have the tests run on this PR and check that the new generated version is ok.

Done!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 636 insertions(+), 93 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 636 insertions(+), 93 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 20 files changed, 788 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics
Total tests: `0`
Passed tests: `0`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>accesscontextmanager</li></ul></blockquote></details>


#### Non-exercised tests

Tests were added that are skipped in VCR:

- TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample

$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/42bb3690-3466-4740-8c88-9ec9181c02f9/build-log/replaying_test.log)

SarahFrench reviewed (commented):
The latest build errors below are probably consequences of deleting the function previously. If you [build the provider locally](https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/) [you can use `make lint`](https://github.com/hashicorp/terraform-provider-google-beta/blob/40747e4195956b3f9f68484ffd30f15595085657/GNUmakefile#L31) in the root of the generated provider repo to get quick feedback about errors like this.

```
# github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accesscontextmanager_test [github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accesscontextmanager.test]
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go:10:2: "github.com/hashicorp/terraform-plugin-sdk/v2/terraform" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go:13:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_ingress_policy_test.go:14:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport" imported as transport_tpg and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:10:2: "github.com/hashicorp/terraform-plugin-sdk/v2/terraform" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:14:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:15:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport" imported as transport_tpg and not used
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 9 files changed, 636 insertions(+), 99 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 9 files changed, 636 insertions(+), 99 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 20 files changed, 788 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics
Total tests: `0`
Passed tests: `0`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>accesscontextmanager</li></ul></blockquote></details>


#### Non-exercised tests

Tests were added that are skipped in VCR:

- TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample
- TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample

$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/32cf7ed9-2de2-40f5-94b5-3384f9baf445/build-log/replaying_test.log)

SarahFrench reviewed (commented):
Here's the current build error:

```
# github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accesscontextmanager_test [github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accesscontextmanager.test]
google-beta/services/accesscontextmanager/resource_access_context_manager_services_perimeters_test.go:52:93: undefined: terraform
google-beta/services/accesscontextmanager/resource_access_context_manager_services_perimeters_test.go:53:17: undefined: terraform
google-beta/services/accesscontextmanager/resource_access_context_manager_services_perimeters_test.go:61:16: undefined: tpgresource
google-beta/services/accesscontextmanager/resource_access_context_manager_services_perimeters_test.go:66:13: undefined: transport_tpg
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:10:2: "github.com/hashicorp/terraform-plugin-sdk/v2/terraform" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:14:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource" imported and not used
google-beta/services/accesscontextmanager/resource_access_context_manager_service_perimeter_test.go:15:2: "github.com/hashicorp/terraform-provider-google-beta/google-beta/transport" imported as transport_tpg and not used
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 636 insertions(+), 100 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 636 insertions(+), 100 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 20 files changed, 788 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: `9`
Passed tests: `1`
Skipped tests: `3`
Affected tests: `5`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>accesscontextmanager</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 5 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample|TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample|TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample|TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample|TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_build/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/9a82c29e-c481-43b9-a5c3-4ff36041b17d/recording) for each test

SarahFrench reviewed (changes requested):
Those test runs have shown that the example files used to generate the tests are flawed in different ways, mostly with fields being repeated in a resource block. The permission error might be a limitation of our test environment, or something we can provision in the test. I'd suggest we tackle the other failures first.


#### TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample 

```
   vcr_utils.go:152: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Too many ingress_from blocks
        
          on terraform_plugin_test.tf line 24, in resource "google_access_context_manager_service_perimeter_ingress_policy" "ingress_policy":
          24:   ingress_from {
        
        No more than 1 "ingress_from" blocks are allowed
--- FAIL: TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample (0.37s)
```

#### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample & TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample
```
2024/06/06 18:43:39 [DEBUG] [transport] [server-transport 0xc002a4a480] loopyWriter exiting with error: transport closed by client 
    vcr_utils.go:152: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating AccessPolicy: googleapi: Error 403: The caller does not have permission
        
          with google_access_context_manager_access_policy.access-policy,
          on terraform_plugin_test.tf line 32, in resource "google_access_context_manager_access_policy" "access-policy":
          32: resource "google_access_context_manager_access_policy" "access-policy" {
        
2024/06/06 18:43:39 [DEBUG] [transport] [server-transport 0xc001f12300] Closing: EOF 
```

#### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample

```
        Error: Attribute redefined
        
          on terraform_plugin_test.tf line 33, in resource "google_access_context_manager_service_perimeter" "test-access":
          33:               identities = ["principal://iam.googleapis.com/locations/global/workforcePools/1234/subject/janedoe"]
        
        The argument "identities" was already set at
        terraform_plugin_test.tf:32,15-25. Each argument may be set only once.
```


#### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample

```
    vcr_utils.go:152: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Unsupported argument
        
          on terraform_plugin_test.tf line 18, in resource "google_access_context_manager_service_perimeters" "secure-data-exchange":
          18:       		vpcAccessibleServices = {
        
        An argument named "vpcAccessibleServices" is not expected here.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 53, in resource "google_access_context_manager_service_perimeter" "test-access":
          53:   parent         = "accessPolicies/${google_access_context_manager_access_policy.test-access.name}"
        
        A managed resource "google_access_context_manager_access_policy"
        "test-access" has not been declared in the root module.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 54, in resource "google_access_context_manager_service_perimeter" "test-access":
          54:   name           = "accessPolicies/${google_access_context_manager_access_policy.test-access.name}/servicePerimeters/%s"
        
        A managed resource "google_access_context_manager_access_policy"
        "test-access" has not been declared in the root module.
        
        Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 69, in resource "google_access_context_manager_service_perimeter" "test-access":
          69: 					access_level = google_access_context_manager_access_level.test-access.name
        
        A managed resource "google_access_context_manager_access_level" "test-access"
        has not been declared in the root module.

```

daanheikens commented:
> Those test runs have shown that the example files used to generate the tests are flawed in different ways, mostly with fields being repeated in a resource block. The permission error might be a limitation of our test environment, or something we can provision in the test. I'd suggest we tackle the other failures first.
> 
> 
> 
> 
> 
> #### TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample 
> 
> 
> 
> ```
> 
>    vcr_utils.go:152: Step 1/2 error: Error running pre-apply refresh: exit status 1
> 
>         
> 
>         Error: Too many ingress_from blocks
> 
>         
> 
>           on terraform_plugin_test.tf line 24, in resource "google_access_context_manager_service_perimeter_ingress_policy" "ingress_policy":
> 
>           24:   ingress_from {
> 
>         
> 
>         No more than 1 "ingress_from" blocks are allowed
> 
> --- FAIL: TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample (0.37s)
> 
> ```
> 
> 
> 
> #### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample & TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample
> 
> ```
> 
> 2024/06/06 18:43:39 [DEBUG] [transport] [server-transport 0xc002a4a480] loopyWriter exiting with error: transport closed by client 
> 
>     vcr_utils.go:152: Step 1/2 error: Error running apply: exit status 1
> 
>         
> 
>         Error: Error creating AccessPolicy: googleapi: Error 403: The caller does not have permission
> 
>         
> 
>           with google_access_context_manager_access_policy.access-policy,
> 
>           on terraform_plugin_test.tf line 32, in resource "google_access_context_manager_access_policy" "access-policy":
> 
>           32: resource "google_access_context_manager_access_policy" "access-policy" {
> 
>         
> 
> 2024/06/06 18:43:39 [DEBUG] [transport] [server-transport 0xc001f12300] Closing: EOF 
> 
> ```
> 
> 
> 
> #### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterGranularControlsExample
> 
> 
> 
> ```
> 
>         Error: Attribute redefined
> 
>         
> 
>           on terraform_plugin_test.tf line 33, in resource "google_access_context_manager_service_perimeter" "test-access":
> 
>           33:               identities = ["principal://iam.googleapis.com/locations/global/workforcePools/1234/subject/janedoe"]
> 
>         
> 
>         The argument "identities" was already set at
> 
>         terraform_plugin_test.tf:32,15-25. Each argument may be set only once.
> 
> ```
> 
> 
> 
> 
> 
> #### TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample
> 
> 
> 
> ```
> 
>     vcr_utils.go:152: Step 1/2 error: Error running pre-apply refresh: exit status 1
> 
>         
> 
>         Error: Unsupported argument
> 
>         
> 
>           on terraform_plugin_test.tf line 18, in resource "google_access_context_manager_service_perimeters" "secure-data-exchange":
> 
>           18:       		vpcAccessibleServices = {
> 
>         
> 
>         An argument named "vpcAccessibleServices" is not expected here.
> 
>         
> 
>         Error: Reference to undeclared resource
> 
>         
> 
>           on terraform_plugin_test.tf line 53, in resource "google_access_context_manager_service_perimeter" "test-access":
> 
>           53:   parent         = "accessPolicies/${google_access_context_manager_access_policy.test-access.name}"
> 
>         
> 
>         A managed resource "google_access_context_manager_access_policy"
> 
>         "test-access" has not been declared in the root module.
> 
>         
> 
>         Error: Reference to undeclared resource
> 
>         
> 
>           on terraform_plugin_test.tf line 54, in resource "google_access_context_manager_service_perimeter" "test-access":
> 
>           54:   name           = "accessPolicies/${google_access_context_manager_access_policy.test-access.name}/servicePerimeters/%s"
> 
>         
> 
>         A managed resource "google_access_context_manager_access_policy"
> 
>         "test-access" has not been declared in the root module.
> 
>         
> 
>         Error: Reference to undeclared resource
> 
>         
> 
>           on terraform_plugin_test.tf line 69, in resource "google_access_context_manager_service_perimeter" "test-access":
> 
>           69: 					access_level = google_access_context_manager_access_level.test-access.name
> 
>         
> 
>         A managed resource "google_access_context_manager_access_level" "test-access"
> 
>         has not been declared in the root module.
> 
> 
> 
> ```

Sure thing! I'll just create a seperate example for this case that would solve it i suppose. Let's see what's left afterwards.

SarahFrench commented:
/gcbrun

daanheikens commented:
> Please address the merge conflicts on this PR- they're preventing automation that runs the tests on PRs. Thanks!

Sorry I've had some busy days which resulted in no response on this PR. I will try to pick this up again asap!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 560 insertions(+), 104 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 560 insertions(+), 104 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 16 files changed, 635 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: 8
Passed tests: 1
Skipped tests: 3
Affected tests: 4

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>accesscontextmanager</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/build-log/recording_build/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/recording/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample.log)]


$\textcolor{red}{\textsf{Errors occurred during RECORDING mode. Please fix them to complete your PR.}}$


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/7b0c98cb-9f22-4478-8f84-9e62a483fe9e/recording) for each test

SarahFrench commented on mmv1/templates/terraform/examples/access_context_manager_service_perimeter_ingress_policy.tf.erb:
```diff
@@ -10,14 +10,17 @@ resource "google_access_context_manager_service_perimeter" "storage-perimeter" {
   perimeter = "${google_access_context_manager_service_perimeter.storage-perimeter.name}"
+
   ingress_from {
     identity_type = "any_identity"
```
```suggestion
    identity_type = "ANY_IDENTITY"
```

SarahFrench commented on mmv1/templates/terraform/examples/access_context_manager_service_perimeter_secure_data_exchange.tf.erb:
```diff
@@ -14,10 +14,11 @@ resource "google_access_context_manager_service_perimeters" "<%= ctx[:primary_re
-		}
+
+			vpc_accessible_services {
+				enableRestriction = true
+				allowedServices = ["bigquery.googleapis.com"]
```
```suggestion
				enable_restriction = true
				allowed_services = ["bigquery.googleapis.com"]
```

SarahFrench reviewed (commented):
Some fixes to error failures - there are some tests failing due to not having permission on org/123456789. I can look into that in future, or you could look at how other YAML files use `test_vars_overrides` to pass different values into documentation versus tests. The tests would need to be passed `"envvar.GetTestOrgFromEnv(t)"`, while documentation can still be passed `"123456789"`

SarahFrench commented:
/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 561 insertions(+), 105 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10806-old..auto-pr-10806) ( 8 files changed, 561 insertions(+), 105 deletions(-))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10806-old..auto-pr-10806) ( 16 files changed, 635 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_access_context_manager_service_perimeter` (14 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_access_context_manager_service_perimeter" "primary" {
  status {
    ingress_policies {
      ingress_from {
        sources {
          resource = # value needed
        }
      }
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: 8
Passed tests: 1
Skipped tests: 3
Affected tests: 4

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>accesscontextmanager</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample</li>
<li>TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/build-log/recording_build/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/recording/TestAccAccessContextManagerServicePerimeterIngressPolicy_accessContextManagerServicePerimeterIngressPolicyExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterBasicExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterDryRunExample.log)]
`TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/build-log/recording_build/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/recording/TestAccAccessContextManagerServicePerimeter_accessContextManagerServicePerimeterSecureDataExchangeExample.log)]


$\textcolor{red}{\textsf{Errors occurred during RECORDING mode. Please fix them to complete your PR.}}$


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10806/artifacts/c83ed985-1913-4593-9b4e-a794ffc754c6/recording) for each test

SarahFrench reviewed (changes requested):
Removed, please see https://github.com/GoogleCloudPlatform/magic-modules/pull/10806#pullrequestreview-2156122025

SarahFrench reviewed (changes requested):
I checked with the core team and acceptance tests for Access Context Manager need to run serially. [This is currently achieved with the tests here](https://github.com/GoogleCloudPlatform/magic-modules/blob/baa571435805ea0fa62fce56aad5230a8533d71f/mmv1/third_party/terraform/services/accesscontextmanager/resource_access_context_manager_access_policy_test.go.erb#L19-L36), and us adding new tests in this PR will interfere.

For us to close out this PR please revert back to your commit updating the field descriptions : https://github.com/GoogleCloudPlatform/magic-modules/pull/10806/commits/bbe195a657d7e7f77939dcf5d7b60173568bd6df

daanheikens commented:
> I checked with the core team and acceptance tests for Access Context Manager need to run serially. [This is currently achieved with the tests here](https://github.com/GoogleCloudPlatform/magic-modules/blob/baa571435805ea0fa62fce56aad5230a8533d71f/mmv1/third_party/terraform/services/accesscontextmanager/resource_access_context_manager_access_policy_test.go.erb#L19-L36), and us adding new tests in this PR will interfere.
> 
> 
> 
> For us to close out this PR please revert back to your commit updating the field descriptions : https://github.com/GoogleCloudPlatform/magic-modules/pull/10806/commits/bbe195a657d7e7f77939dcf5d7b60173568bd6df

Hi Sarah, after reading this I believe i misunderstood your comment. I reverted the actual commit, but i believe you meant we have to revert the branch back to that particular commit to remove all the changes after that one, correct?

Please confirm if so then i will correct my change.

SarahFrench commented:
@daanheikens - Sorry for causing confusion, I meant that this PR should be returned to only making documentation updates and shouldn't try to add new acceptance tests.

daanheikens commented:
@SarahFrench i think we can close this one, it's already being pushed to main via another way I see looking at the merge conflict. That means this PR is obsolete!

SarahFrench reviewed (approved):
Thanks!

SarahFrench commented:
From the merge conflicts I can see this PR was superseded by https://github.com/GoogleCloudPlatform/magic-modules/pull/11273/files
